mgmtsystem/ec2.py

```diff
@@ -450,16 +455,29 @@ def remove_host_from_cluster(self, hostname):
 
     def create_s3_bucket(self, bucket_name):
         self.logger.info("Creating bucket: {}".format(bucket_name))
-        self.s3_connection.create_bucket(bucket_name)
+        try:
+            self.s3_connection.create_bucket(Bucket=bucket_name, CreateBucketConfiguration={
+                'LocationConstraint': self.kwargs.get('region')})
+            self.logger.info("Success: Bucket was successfully created.")
+            return True
+        except Exception:
+            self.logger.exception("Error: Bucket was not successfully created.")
+            return False
 
-    def upload_file_to_s3_bucket(self, bucket_name, file_path, key_name):
-        bucket = self.s3_connection.get_bucket(bucket_name)
+    def upload_file_to_s3_bucket(self, bucket_name, file_path, file_name):
+        bucket = self.s3_connection.Bucket(bucket_name)
         self.logger.info("uploading file {} to bucket: {}".format(file_path, bucket_name))
-        key = boto.s3.key.Key(bucket, key_name)
-        with open(file_path) as f:
-            key.set_contents_from_file(f)
-        self.logger.info("Success: uploading file completed")
-        return True
+        if os.path.isfile(file_path):
+            try:
+                bucket.upload_file(file_path, file_name)
+                self.logger.info("Success: uploading file completed")
+                return True
+            except Exception:
```

Comment:
Actually including the exception, formatted as a string (or using ex.message directly), will greatly help us debug when an upload does fail, without re-running in a debug mode.